wave_matching/include/wave/matching/ndt.hpp

```diff
@@ -0,0 +1,29 @@
+// Class to wrap PCL ICP matcher
+
+#ifndef WAVE_MATCHING_NDT_HPP
+#define WAVE_MATCHING_NDT_HPP
+
+#include <pcl/point_types.h>
+#include <pcl/point_cloud.h>
+#include <pcl/registration/ndt.h>
+#include "wave/matching/matcher.hpp"
+#include "wave/matching/pcl_common.hpp"
+
+namespace wave {
+
+class NDTMatcher : public Matcher<PCLPointCloud> {
+ public:
+    explicit NDTMatcher(float resolution, const std::string &config_path);
```

Comment:
Not required, passing by value for built-in types are generally preferred, further const does not make sense because you are passing by value, `const` keyword does nothing.